sys/mips/mips/pm_machdep.c

```diff
@@ -567,10 +571,7 @@ set_mcontext(struct thread *td, mcontext_t *mcp)
 		bcopy((void *)&mcp->mc_fpregs, (void *)&td->td_frame->f0,
 		    sizeof(mcp->mc_fpregs));
 	}
-#if __has_feature(capabilities)
-	td->td_frame->pc =
-	    update_pcc_offset(mcp->mc_cheriframe.cf_pcc, mcp->mc_pc);
-#else
+#if !__has_feature(capabilities)
```

Comment:
Maybe hoist the
```
#if __has_feature(capabilities)
	cheri_trapframe_from_cheriframe(tp, &mcp->mc_cheriframe);
#endif
```
into an `#else` to reduce the number of ifdefs?

Reply:
Actually, I need to move it earlier to deal with a bug we have now where the GET_MC_CLEAR_RET case doesn't actually clear c3 like it should.